shared_storage/sharedstorage.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+ * Author: BogDan Vatra <bogdan.vatra@kdab.com>
+ * SPDX-License-Identifier: MIT
+*/
+#pragma once
+
+#include <QObject>
+#include <functional>
+
+#include "utils.h"
+
+namespace android {
+namespace provider {
```

Comment:
assuming C++17, this could read `namespace android::provider {` (also elsewhere)

Reply:
Qt 5.15 is built with C++14 ...